node/node_handler.go

```diff
@@ -431,19 +433,33 @@ func (node *Node) SendPongMessage() {
 						utils.GetLogInstance().Info("[PONG] sent pong message to", "group", node.NodeConfig.GetShardGroupID(), "# nodes", numPeersNow)
 					}
 					sentMessage = true
-					// stop sending ping message
-					node.serviceManager.TakeAction(&service.Action{Action: service.Stop, ServiceType: service.PeerDiscovery})
 					// wait a bit until all validators received pong message
 					time.Sleep(200 * time.Millisecond)
+
 					// only need to notify consensus leader once to start the consensus
 					if firstTime {
+						// Leader stops sending ping message
+						node.serviceManager.TakeAction(&service.Action{Action: service.Stop, ServiceType: service.PeerDiscovery})
```

Comment:
Why the leader stops sending ping here?

Reply:
main purpose of ping message is to send pubkey.  Leader's pubkey will be sent in pong message.  I think there is no need for leader to keep sending ping message then.